layers.go

```diff
@@ -486,7 +486,11 @@ func newLayerStore(rundir string, layerdir string, driver drivers.Driver, uidMap
 	if err != nil {
 		return nil, err
 	}
-	lockfile.Lock()
+	if s.readOnlyLayers {
+		lockfile.RLock()
+	} else {
+		lockfile.Lock()
+	}
```

Comment:
I wonder why we to lock here at all. We're only calling public APIs; `GetLockfile` serializes own its own and `(*layerStore) Load` is locking as well. I think we can ditch the locking here entirely; @nalind PTAL.

Reply:
There's also `newROLayerStore`: Why not use that?